### PR TITLE
TEST-#6830: Use local s3 server instead of public s3 buckets

### LINKED ISSUE #6830: Remove access to public s3 buckets
**Is your feature request related to a problem? Please describe.**
This is a compliance issue which comes from a principle of least privilege.

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6830 
- [x] tests ~added and~ passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/test/test_headers.py [hunk 1]
```diff
@@ -44,6 +44,8 @@ def test_line_endings():
         if any(i in subdir for i in [".git", ".idea", "__pycache__"]):
             continue
         for file in files:
+            if file.endswith(".parquet"):
```
Comment (anmyachev) [inline modin/test/test_headers.py:47]: In binary format there is no need to check these characters (`\r\n`).

File: modin/core/io/text/json_dispatcher.py [hunk 2]
```diff
@@ -60,12 +62,21 @@ def _read(cls, path_or_buf, **kwargs):
             return cls.single_worker_read(
                 path_or_buf, reason="`lines` argument not supported", **kwargs
             )
-        with OpenFile(path_or_buf, "rb") as f:
+        with OpenFile(
+            path_or_buf,
+            "rb",
+            **(kwargs.get("storage_options", None) or {}),
```
Comment (anmyachev) [inline modin/core/io/text/json_dispatcher.py:68]: Use of `storage_options` parameter was missed

File: modin/core/io/text/json_dispatcher.py [hunk 3]
```diff
@@ -60,12 +62,21 @@ def _read(cls, path_or_buf, **kwargs):
...
             columns = pandas.read_json(BytesIO(b"" + f.readline()), lines=True).columns
         kwargs["columns"] = columns
         empty_pd_df = pandas.DataFrame(columns=columns)
 
-        with OpenFile(path_or_buf, "rb", kwargs.get("compression", "infer")) as f:
+        with OpenFile(
+            path_or_buf,
+            "rb",
+            kwargs.get("compression", "infer"),
+            **(kwargs.get("storage_options", None) or {}),
```
Comment (anmyachev) [inline modin/core/io/text/json_dispatcher.py:78]: Use of `storage_options` parameter was missed

File: modin/experimental/core/io/text/csv_glob_dispatcher.py [hunk 4]
```diff
@@ -68,7 +68,9 @@ def _read(cls, filepath_or_buffer, **kwargs):
                     reason=cls._file_not_found_msg(filepath_or_buffer),
                     **kwargs,
                 )
-            filepath_or_buffer = cls.get_path(filepath_or_buffer)
+            filepath_or_buffer = cls.get_path(
+                filepath_or_buffer, kwargs.get("storage_options")
```
Comment (anmyachev) [inline modin/experimental/core/io/text/csv_glob_dispatcher.py:72]: Use of `storage_options` parameter was missed

File: modin/experimental/core/io/text/csv_glob_dispatcher.py [hunk 5]
```diff
@@ -363,11 +367,17 @@ def get_file_path(fs_handle) -> List[str]:
...
             return fs_addresses
 
-        fs, _ = fsspec.core.url_to_fs(file_path)
+        if storage_options is not None:
+            new_storage_options = dict(storage_options)
+            new_storage_options.pop("anon", None)
+        else:
+            new_storage_options = {}
+
+        fs, _ = fsspec.core.url_to_fs(file_path, **new_storage_options)
```
Comment (anmyachev) [inline modin/experimental/core/io/text/csv_glob_dispatcher.py:376]: Made by analogy with other functions

**Comment by anmyachev** [comment]
@YarShev @dchigarev ready for review

**Review by YarShev — APPROVED** [review]

File: modin/test/test_headers.py (hunk 1 above)
Comment (anmyachev) [reply modin/test/test_headers.py:47]: This sequence of characters may have a different meaning for binary formats. If, for example, you try to replace `\r\n` with `\n`, the arrow will give a message that parquet file is corrupted.

File: modin/experimental/pandas/test/test_io_exp.py [hunk 6]
```diff
@@ -212,17 +216,19 @@ def _pandas_read_csv_glob(path, storage_options):
     reason=f"{Engine.get()} does not have experimental API",
 )
 @pytest.mark.parametrize(
-    "storage_options",
-    [{"anon": False}, {"anon": True}, {"key": "123", "secret": "123"}, None],
```
[the remark this answers is not in the document]
Comment (YarShev) [reply modin/experimental/pandas/test/test_io_exp.py:216]: What if we want to add a test case for `storage_options=None` in future? What are we supposed to do?

[the remark this answers is not in the document]
Comment (anmyachev) [reply modin/experimental/pandas/test/test_io_exp.py:216]: At least, do not use your our own s3 server, since in this case we always need to explicitly specify endpoint.
